### PR TITLE
Restructure style dictionary tokens, add extra mappings

### PR DESCRIPTION
**Pull request by brecht**

### CODE REVIEW

**Comment by thibmaek** [comment]
I have tagged this as milestone 4.0.0 since it includes a breaking change in API for Style Dictionary output @brecht

**Review by sHoskens — APPROVED** [review]

**Review by Stvenoo — APPROVED** [review]

File: lib/font.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  book: 350, // exceptional, book variants should not be used for devices, but just in case
+  normal: 400,
+  medium: 500,
+  semiBold: 600,
+  bold: 700,
+  extraBold: 800,
+  black: 900,
+};
+
+const fontWeights = Object.keys(FONT_WEIGHT).map(weightId => ({
```
Comment (thibmaek) [inline lib/font.js:17]: Why not just create an array, to begin with? Then the mapping is not needed and the order is guaranteed.

File: lib/font.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  name: weightId,
+  value: FONT_WEIGHT[weightId],
+}));
+
+const getFontWeight = (fontWeightName) => {
+  const key = camelCase(fontWeightName);
+  return typeof(FONT_WEIGHT[key]) !== 'undefined' ? FONT_WEIGHT[key] : FONT_WEIGHT.normal;
```
Comment (thibmaek) [inline lib/font.js:24]: Wow did not know typeof was a function!

File: lib/styleDictionary/assetMapping.js [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
 const mapTextStyleAssets = (textStyles = []) => {
   const output = createTreeOfLeastDepth(
-    textStyles,
-    ['id'],
-    token => ({ value: `${token.family}` })
+    uniqueArrayByProperty(textStyles, 'family'),
+    ['family'],
+    token => ({
+      name: { value: token.family },
+      // woff2: { value: 'path/to/font.woff2' },
```
Comment (thibmaek) [inline lib/styleDictionary/assetMapping.js]: still needed?

File: lib/styleDictionary/util.js [hunk 4]
```diff
@@ -77,8 +77,50 @@ const createTreeOfLeastDepth = (inputTokens, identifiers, tokenConverter) => {
   }, {});
 };
 
+/**
+ * Extracts a font family's base name, e.g. 'Roboto-Regular' => 'Roboto'
+ */
+const getBaseNameFromFontFamily = family => family.split('-')[0];
```
Comment (thibmaek) [inline lib/styleDictionary/util.js]: What if the font is RobotoRegular.ttf?

Comment (brecht) [reply lib/styleDictionary/util.js]: convention :)

File: lib/styleDictionary/assetMapping.js (hunk 3 above)
Comment (brecht) [reply lib/styleDictionary/assetMapping.js]: nah, will remove

File: lib/font.js (hunk 1 above)
Comment (brecht) [reply lib/font.js:17]: see e.g. getFontWeight(), I think using FONT_WEIGHT.normal is clearer than fontWeights[4]

File: lib/styleDictionary/util.js (hunk 4 above)
Comment (brecht) [reply lib/styleDictionary/util.js]: removed this functionality, since it's only relevant for web, will move it to the rendering target mapping

**Review by thibmaek — APPROVED** [review]

**Pull request merged**